### PR TITLE
Contribution graph with multiple candidates

### PR DESCRIPTION
**Pull request by tingaloo**
We can view multiple candidate campaign contributions within the same graph.

## Description
Whilst in the selection screen with AT LEAST 2 CANDIDATES, a button will appear to compare multiple candidates. The user can click the button and be routed to a new graph page, looking like the screenshots below.

Note: There are no other constraints such as campaigns must be within the same year, so comparisons can become wild depending on the difference in time/contribution amounts.

Also note: this code can always be optimized, feel free to wreck it later.

## Motivation and Context
Compare candidate contributions on the same graph.

## How Has This Been Tested?
Tested locally.

## Screenshots (highly recommended):

A fair omparison.
screen shot 2016-06-19 at 6 47 20 pm

A little more wild..
screen shot 2016-06-19 at 6 51 48 pm

Button displays when user selects AT LEAST 2 CANDIDATES, meaning multiple campaigns from a single candidate isn't enough.

screen shot 2016-06-19 at 7 01 12 pm

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I've talked through the changes with a team lead
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/components/graphs/contributions/chartWrapper.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      candidates: this.props.candidates
+    }
+  }
+
+  ChartFactory (type, el) {
+      switch (type) {
+          case 'default':
+              return new GraphDefault(el, this.getChartState());
+          default:
+            console.log("foo")
```
Comment (mkalish) [inline src/components/graphs/contributions/chartWrapper.js]: Let's pull this console statement out

File: src/components/graphs/contributions/contributionsGraphContainer.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        candidates = this.props.location.state,
+        chartData = this.state.chartData
+
+    let candidateCampaigns = candidates.map((candidate) => {
+      for (let campaign of candidate.data.campaigns) {
+        return <p> {candidate.candidateName} - {campaign.raceType} </p>
+      }
+    })
+    // check for data integrity
+    if (chartData.length == candidates.length) {
```
Comment (mkalish) [inline src/components/graphs/contributions/contributionsGraphContainer.js]: `===` for safety sake

**Comment by mkalish** [comment]
This looks awesome... there are a few conflicts so if you re-up from develop and get those merged in; we'll get this in for when I push code next

**Comment by tingaloo** [comment]
Good catches, all patched up.

**Pull request merged**